python/cudf/utils/cudautils.py

```diff
@@ -795,6 +796,102 @@ def gpu_mark_seg_segments(begins, markers):
         markers[begins[i]] = 1
 
 
+# taken from the following repo:
+# https://github.com/numba/numba-examples/
+@cuda.jit
+def gpu_min_max(arr, min_max_array):
+    start = cuda.grid(1)
+    stride = cuda.gridsize(1)
+
+    # Array already seeded with starting values appropriate for x's dtype
+    # Not a problem if this array has already been updated
+    local_min = min_max_array[0]
+    local_max = min_max_array[1]
+
+    for i in range(start, arr.shape[0], stride):
+        element = arr[i]
+        local_min = min(element, local_min)
+        local_max = max(element, local_max)
+
+    # Now combine each thread local min and max
+    cuda.atomic.min(min_max_array, 0, local_min)
+    cuda.atomic.max(min_max_array, 1, local_max)
+
+
+def min_max(arr):
+    dtype_min, dtype_max = dtype_min_max(arr.dtype)
+    min_max_array_gpu = rmm.device_array(shape=(2,), dtype=arr.dtype)
+    min_max_array_gpu[0] = dtype_max
+    min_max_array_gpu[1] = dtype_min
+    gpu_min_max.forall(arr.size)(arr, min_max_array_gpu)
+    min_value = min_max_array_gpu[0]
+    max_value = min_max_array_gpu[1]
+    return min_value, max_value
+
+
+@cuda.jit
+def gpu_mark_found_int(arr, val, out):
+    i = cuda.grid(1)
+    if i < arr.size:
+        if check_equals_int(arr[i], val):
+            out[i] = i
```

Comment:
Do you need and else case of setting to 0 here since the memory is potentially uninitialized?

Reply:
Thanks for catching this.

Right now I'm making sure to populate `arr` with some value outside the range `[0, arr.size)` before calling this kernel. I think that's awkward and will figure out something better here.

Reply:
I've addressed this by requiring an additional argument `not_found` which is the value to be set if the element is not found.